src/System.Management.Automation/engine/Modules/ImportModuleCommand.cs

```diff
@@ -1941,6 +1962,56 @@ private List<T> FilterModuleCollection<T>(IEnumerable<T> moduleCollection)
             return filteredModuleCollection;
         }
 
+        private void PrepareNoClobberWinCompatModuleImport(string moduleName, ModuleSpecification moduleSpec, ref ImportModuleOptions importModuleOptions)
+        {
+            List<string> NoClobberModuleList = PowerShellConfig.Instance.GetWindowsPowerShellCompatibilityNoClobberModuleList();
+
+            // moduleName can be just a module name and it also can be a full path to psd1 from which we need to extract the module name
+            var coreModuleToLoad = moduleName;
```

Comment:
Module specs might also have the name field be a path, which we support, since `RequiredModules` depends on it.

```suggestion
            string coreModuleToLoad = moduleSpec?.Name ?? moduleName;
            if (Path.IsPathRooted(coreModuleToLoad)) // Also need to deal with relative paths here -- helper function?
            {
                coreModuleToLoad = Path.GetFileNameWithoutExtension(coreModuleToLoad);
            }
```

Also, what's the behaviour here for relative paths?

Reply:
Updated to use `ModuleIntrinsics.GetModuleName` and use it for `moduleSpec.Name` as well.